### PR TITLE
genesys: trivial rework some usbhub errors

### PR DESCRIPTION
**Pull request by gportay**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

The idea behind those changes is to prepare the genesys plugin for upgrading the genuine Genesys Logic Hubs products.

These products uses `USB\VID_05E3&PID_0610` which is not set in the quirk file yet; will be in a dedicated PR, once the support is fully functional.

The support for genuine Genesys Logic Hubs is splited into several PR. This one is the second of the serie.

In short, this PR:
 - Print the Genesys IC if it is not yet supported by the plugin.
 - Fails silently if the IC is not yet supported.

### CODE REVIEW

**Comment by gportay** [comment]
Tested with a hub found in my stuff that is based on the `GL33521-22`.

**Comment by gportay** [comment]
@adamgene, do you have any thing else to add?

**Review by adamgene — CHANGES_REQUESTED** [review]
```
	if (memcmp(&self->static_ts.tool_string_version, "USB", 3) == 0) {
		g_set_error(error,
			    FWUPD_ERROR,
			    FWUPD_ERROR_INTERNAL,
			    "Unsupported IC: tool string unsupported");
		return FALSE;
	}
```
Can you add these code @741? I found out old GL3520 unsupported tool string.

File: plugins/genesys/fu-genesys-usbhub-device.c [hunk 1]
```diff
@@ -746,10 +748,19 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
...
-					    FWUPD_ERROR,
-					    FWUPD_ERROR_INTERNAL,
-					    "Unknown ISP model");
+			g_autofree gchar *ic_type =
+			    fu_common_strsafe((const gchar *)self->static_ts.mask_project_ic_type,
+					      6);
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "Unsupported IC GL%c%c%c%c-%c%c",
```
Comment (hughsie) [inline plugins/genesys/fu-genesys-usbhub-device.c:757]: I don't think you know that strlen(ic_type) >= 6 -- I think you either want to print this as %s (probably fine given it's in the error path) or make a `convert_mask_project_ic_type_to_string` function that adds the dash in the right place.

Comment (gportay) [reply plugins/genesys/fu-genesys-usbhub-device.c:757]: make sense.
done

**Comment by gportay** [comment]
> ```
> 	if (memcmp(&self->static_ts.tool_string_version, "USB", 3) == 0) {
> 		g_set_error(error,
> 			    FWUPD_ERROR,
> 			    FWUPD_ERROR_INTERNAL,
> 			    "Unsupported IC: tool string unsupported");
> 		return FALSE;
> 	}
> ```
> 
> Can you add these code @741? I found out old GL3520 unsupported tool string.

I will do it in another PR, as it does not appear to be directly linked to that one.

**Comment by gportay** [comment]
@hughsie How can I re-trigger the CI? I did not understand what happened.

Ready to get merged otherwise.

**Review by hughsie — APPROVED** [review]
